### PR TITLE
Initial commit for asciidoc contribution layer for Spacemacs

### PR DESCRIPTION
**Pull request by lehoff**
Rudimentary contribution layer for asciidoc based on adoc-mode.

Main achievement is that it loads adoc-mode.
Only very few keybindings defined.

### CODE REVIEW

**Comment by cpaulik** [comment]
It would be great if the keybindings could follow the [Conventions for markup languages](https://github.com/syl20bnr/spacemacs/blob/develop/doc/CONVENTIONS.md#plain-text-markup-languages). 

You might want to coordinate with https://github.com/syl20bnr/spacemacs/pull/1650

**Comment by lehoff** [comment]
I have done a fix - should I just retract this PR and create a new one?
If not, how do I update this PR with my new files?

**Comment by TheBB** [comment]
The PR is tied to your branch. If you update your branch (which you have), you don't have to do anything else.

**Comment by lehoff** [comment]
@TheBB: thx for the info! Hope my changes are good enough. A bit in doubt with the keybindings.el file and the set-global-key.

File: contrib/lang/asciidoc/keybindings.el [hunk 1]
```diff
@@ -0,0 +1,2 @@
+(global-set-key (kbd "M-h") 'adoc-promote)
```
Comment (cpaulik) [inline contrib/lang/asciidoc/keybindings.el]: These should not be set global

you should use something like

`(define-key adoc-mode-map (kbd "M-h") 'adoc-promote)

although the adoc mode map might be called differently

**Comment by TheBB** [comment]
Yeah, you don't want the global keybindings. They should probably go in `adoc-mode-map`.

**Comment by hijarian** [comment]
This is just unfair. I have opened the PR with the same package 4 days ago and got no response. This PR was added 4 *hours* ago and already has contribution! :E

**Comment by hijarian** [comment]
:'(

**Comment by cpaulik** [comment]
@hijarian Yeah I'm sorry. I looked at your PR a few days ago to check the keybindings but since you wrote there are none I did not investigate further.

**Comment by hijarian** [comment]
@cpaulik Anyway. This layer has 2 important features my layer doesn't have:

1. It defines keybindings, albeit rudimentary ones. Printing `0i= ` is simpler than `<SPC>mh1` IMHO and doesn't requre remembering Yet Another Key Sequence. Keybindings like interactive "insert-image" or, say, "table-cell-align-right" would be a lot more useful.
2. It requires `'cl`, which I promptly forgot. :D Sorry about that.

It has the following problem: 

1. It does not follow [contribution guidelines](https://github.com/syl20bnr/spacemacs/blob/master/doc/CONTRIBUTE.md#submitting-a-configuration-layer) Starting from the fact it's pulled against `master`. I don't know, though, whether we actually follow the guidelines here.

**Comment by hijarian** [comment]
@lehoff Keybindings block and `'cl` is simpler to add than to re-start your PR to follow the contribution guidelines. Let's consolidate this layer in #1650. I'll add the missing pieces to my branch and re-make PR again, and you'll remove this PR altogether and will pull my branch while this PR is under review?

**Comment by cpaulik** [comment]
@hijarian I didn't even realize that is was against master :)

I find the keybindings for markup languages useful since I write in several and don't remember in every markup how the exact formatting is supposed to be. So I prefer pressing `SPC m h 1` or `SPC m x b` and get the desired result in every language. At least that is the goal.

**Comment by TheBB** [comment]
I don't think it's a big problem that it's on master: it's a new layer. And in any case, @syl20bnr only every cherry picks.

**Comment by hijarian** [comment]
@cpaulik Oh, that's a valid point! Then we need even more keybindings, at least the same number as in Markdown layer. Need to re-read the code of adoc-mode to extract the corresponding functions, though.

**Comment by hijarian** [comment]
I have re-made the PR as the PR #1697. This one can be safely closed now. @lehoff is written among authors.

**Comment by syl20bnr** [comment]
Thank you for the PR, closed in favor of #1697

**Pull request closed without merging**